src/main/java/MegaMillions.java

```diff
@@ -58,26 +55,45 @@ public void addMegaBalls(String line, List<Integer> megaBalls){
         }
     }
 
-    private List<Number> getUniqueNumbersWithOccurs(List<Integer> list) {
-        List<Number> numbersWithOccurs = new ArrayList<>();
-        for (int i = 1; i <= 10000; i++) {
+    private TreeMap<String, Integer> getUniqueNumbersWithOccurs(List<Integer> list) {
+        TreeMap<String, Integer> numbersWithOccurs = new TreeMap<>();
+        for (int i = 1; i <= Collections.max(list); i++) {
             int occurs = 0;
             for (int j = 0; j < list.size(); j++) {
                 if (list.get(j) == i) {
                     occurs++;
                 }
             }
-            numbersWithOccurs.add(new Number(i, occurs));
+            numbersWithOccurs.put(String.valueOf(i), occurs);
         }
         return numbersWithOccurs;
     }
 
-    public void mostPopularNumberFromList(List<Number> numbers, int howManyPrint) {
-        numbers.sort(Comparator.comparingInt(Number::getHowManyTimesOccurs).reversed());
-        for (int i = 0; i < howManyPrint; i++) {
-            String info = numbers.get(i).toString();
-            System.out.println(info);
+    public void mostPopularNumberFromList(TreeMap<String, Integer> numbers, int howManyPrint) {
+        LinkedHashMap<String, Integer> hashMap = numbers.entrySet().stream().sorted(Comparator.comparing(
```

Comment:
hmmm this all seems more complicated than it should be. Maybe your first solution was easier. I forgot that we cannot get elements from map by their index :/